boards: olimex_lora_stm32wl_devkit: Support newer board revisions

knthm opened the pull request:
Add configuration and DT overlays for new devkit revisions C and D.

This enables identical behavior to earlier revisions by powering the VDDIO pins on the UEXT1 connector and CON1 header via the new FET2 regulator. In these new revisons this regulator is also user-togglable.

Update documentation build/flash/debug steps.

---

The C and D board revisions add a MOSFET for switching the VDDIO supply on the UEXT1 connector and CON1 header (as well as the CDS cell voltage divider):

![Rev-CD-FET2](https://user-images.githubusercontent.com/6822406/228614241-77799506-f448-4473-abf2-dfbac0f04fab.jpg)

I've tested this PR's functionality and confirm it working on my revision D devkit.

knthm committed:
boards: olimex_lora_stm32wl_devkit: Support newer board revisions

Add configuration and DT overlays for new devkit revisions C and D.

This enables identical behavior to earlier revisions by powering the VDDIO
pins on the UEXT1 connector and CON1 header via the new FET2 regulator.
In these new revisons this regulator is also user-togglable.

Update documentation build/flash/debug steps.

Signed-off-by: Kenneth J. Miller <ken@miller.ec>

KloudJack commented:
- Fix formatting snafus (e.g. spaces vs tabs and double tabs)
- Reworded commit message for conciseness

knthm committed:
boards: olimex_lora_stm32wl_devkit: Remove non-applicable LPUART pins

Remove CTS/RTS LPUART pin definitions, as these aren't exposed through
the UEXT1 connector with their accompanying Tx/Rx lines, but are available
separately as GPIO pins on the CON1 pin header.

Signed-off-by: Kenneth J. Miller <ken@miller.ec>

KloudJack commented:
- Updated original commit removing LPUART definition changes from revision C/D overlays
- Added new commit removing non-applicable LPUART lines for all board revisions

KloudJack commented:
Unit test for the STM32 hwinfo driver are failing since I've rebased on main with the above requested changes, seems to trace back to #55993.

This unit test failure is orthogonal to the changes in this PR.

KloudJack commented:
- Rebase on main
